ReactNativeClient/lib/QuickActions.js

```diff
@@ -0,0 +1,45 @@
+
+import {DeviceEventEmitter} from 'react-native';
+import QuickActions from 'react-native-quick-actions';
+const { _ } = require('lib/locale.js');
+
+export default (dispatch, folderId) => {
+	QuickActions.setShortcutItems([
+		{type: 'New note', title: _('New note'), icon: 'Compose'},
+		{type: 'New to-do', title: _('New to-do'), icon: 'Add'},
+	]);
+
+	DeviceEventEmitter.addListener('quickActionShortcut', data => {
+		// This dispatch is to momentarily go back to reset state, similar to what
+		// happens in onJoplinLinkClick_(). Easier to just go back, then go to the
+		// note since the Note screen doesn't handle reloading a different note.
+		//
+		// This hack is necessary because otherwise you get this problem:
+		// The first time you create a note from the quick-action menu, it works
+		// perfectly. But if you do it again immediately later, it re-opens the
+		// page to that first note you made rather than creating an entirely new
+		// note. If you navigate around enough (which I think changes the redux
+		// state sufficiently or something), then it'll work again.
+		dispatch({type: 'NAV_BACK'});
+
```

Comment:
Please could you move this file outside of "lib"?

Also would you be ok with converting it to TypeScript? I think it should be relatively easy if you follow these steps:

- Rename the file to QuickAction.ts
- Add the generated "QuickAction.js" to .gitignore and .eslintignore
- Run `npm run tsc` from the root, which might display a few errors related to types. I don't know if you are familiar with TypeScript, but most likely to set the type of the function parameters. So `dispatch:Function`, `folderId:string`, `data:any`. Then try again `npm run tsc`.

Other than that, I think the file should pass the TypeScript compilation, if not please let me know.

Reply:
> Please could you move this file outside of "lib"?

Sure thing! I just moved it up one level, under `ReactNativeClient/`.

> Also would you be ok with converting it to TypeScript?

Yep, just pushed that change and after some fiddling it seems to compile just fine. Excited to see the project transitioning towards Typescript!